### DISCUSSION TITLE
NiceGUI SQLite example very slow

### DISCUSSION THREAD

**Discussion by panaczech** [zauberzeug/nicegui#2325] [Q&A] [answered]
### Question

Hi,
First I'm sorry that I was unable to dig deeper, I'm learning both NiceGUI and Tortoise ORM.

I was testing some of the examples from this repository and wanted to build on this particular one: https://github.com/zauberzeug/nicegui/tree/main/examples/sqlite_database

It works fine until there is let's say 50-100 records, then it get's laggy. The GUI disappears for several seconds when the ui.input loses focus or when a record is deleted. What causes this behavior? Is there possibly a workaround?

https://github.com/zauberzeug/nicegui/assets/7826307/49e67817-c239-40f8-b344-c57b900c451e

NiceGUI Version: 1.4.5
Python 3.8.1
Linux Mint 20.1 Cinnamon
Intel© Core™ i7-9750HF CPU @ 2.60GHz × 6

**Comment by rodja** [comment, marked as answer]
This example only works for a small number of elements. If you have lots of entries you should restrict/filter the number of elements to show. No user wants to browse 500+ of these cards.

**Reply by falkoschindler** [reply]
Especially re-rendering _all_ cards using `ui.refreshable` isn't very effective for a large number of entries.

**Reply by panaczech** [reply]
Thank you both for the quick answer 👍
